protographic/src/operations/proto-text-generator.ts

```diff
@@ -0,0 +1,363 @@
+import protobuf from 'protobufjs';
+
+/**
+ * Extended Method interface that includes custom properties
+ */
+interface MethodWithIdempotency extends protobuf.Method {
+  idempotencyLevel?: 'NO_SIDE_EFFECTS' | 'DEFAULT';
+}
+
+/**
+ * Options for generating proto text
+ */
+export interface ProtoTextOptions {
+  /** Package name for the proto file */
+  packageName?: string;
+  /** Go package option */
+  goPackage?: string;
+  /** Java package option */
+  javaPackage?: string;
+  /** Java outer classname option */
+  javaOuterClassname?: string;
+  /** Java multiple files option */
+  javaMultipleFiles?: boolean;
+  /** C# namespace option */
+  csharpNamespace?: string;
+  /** Ruby package option */
+  rubyPackage?: string;
+  /** PHP namespace option */
+  phpNamespace?: string;
+  /** PHP metadata namespace option */
+  phpMetadataNamespace?: string;
+  /** Objective-C class prefix option */
+  objcClassPrefix?: string;
+  /** Swift prefix option */
+  swiftPrefix?: string;
+  /** Additional imports to include */
+  imports?: string[];
+  /** Additional options to include */
+  options?: string[];
+  /** Whether to include comments */
+  includeComments?: boolean;
+}
```

Comment:
This structure or something similar has been declared multiple times already. This is a bit confusing.